tpm1_cmds.c

```diff
@@ -64,69 +64,29 @@ int tpm1_pcr_extend(struct tpm *t, struct tpm_digest *d)
 
 	hdr->size = cpu_to_be32(tpmb_size(b));
 
-	switch (t->intf) {
-	case TPM_DEVNODE:
-		/* Not implemented yet */
-		ret = -EBADRQC;
-		break;
-	case TPM_TIS:
-		if (be32_to_cpu(hdr->size) != tis_send(b))
-			ret = -EAGAIN;
-		break;
-	case TPM_CRB:
-		/* Not valid for TPM 1.2 */
-		ret = -ENODEV;
-		break;
-	case TPM_UEFI:
-		/* Not implemented yet */
-		ret = -EBADRQC;
-		break;
-	}
-
-	if (ret)
+	if (be32_to_cpu(hdr->size) != t->ops.send(b)) {
+		ret = -EAGAIN;
 		goto free;
-
-	tpmb_free(b);
+	}
 
 	/* Reset buffer for receive */
-	if (!tpmb_reserve(b)) {
-		ret = -ENOMEM;
-		goto out;
-	}
+	tpmb_trim(tpmb_size(b));
```

Comment:
I guess trim just resets the whole buffer if you trim the entire thing?

Reply:
The trim method shrinks the buffer usage, so this should trim back to the beginning if I didn't make a mistake elsewhere in the put method.